server/jetstream.go

```diff
@@ -223,7 +244,7 @@ func (s *Server) enableJetStream(cfg JetStreamConfig) error {
 
 	// FIXME(dlc) - Allow memory only operation?
 	if stat, err := os.Stat(cfg.StoreDir); os.IsNotExist(err) {
-		if err := os.MkdirAll(cfg.StoreDir, 0755); err != nil {
+		if err := os.MkdirAll(cfg.StoreDir, 0750); err != nil {
```

Comment:
This one looks like a good change but makes me nervous for a dot release.

Interested in what @kozlovic thinks and possibly @ripienaar

Reply:
I really don't know the consequences of this change.

Reply:
Also agree in principal but feel we need to consider it a bit more and test more especially to see how this interacts on windows.

Reply:
Can remove the 0755->0750 change if desired

Reply:
Yeah, go does claim to do the right thing - but they are known for doing clever but dumb things rather than clever but right things (http headers, arbitrary x509 restrictions etc) so I'd rather trust but verify that their clever thing is actually the right thing.

Also being that we now made it 755 if we change it we should probably make this a configurable to not burn users who now depend on the past behaviour.

Reply:
Yeah, I wondered about making it configurable.  I think we have the fire of the current issue and the desire to fix the permissions and should avoid compounding issues.  I've undone the permissions change and stuck to the same 0755 used in the old code and the new.

Reply:
https://github.com/nats-io/nats-server/issues/2217 filed to track the permissions change.  Resolving this conversation.